crawl-ref/source/dat/des/variable/large_themed.des

```diff
@@ -2316,3 +2316,85 @@ xxxx-xx..xx....xx-xxx-x
 xxxxxxxxx-x.....xxxxxxx
 xxxxxxxxxxx.@@@..xxxxxx
 ENDMAP
+
+NAME:   dreamdust_cave_troll
+TAGS:   no_monster_gen
+ORIENT: northwest
+DEPTH:  D:10-, Lair
+NSUBST: - = 1 / 2=? / .
+KFEAT:  ^ = web
+MONS:   deep troll
+: dgn.delayed_decay(_G, "?", "yak corpse")
+MARKER: ! = lua:fog_machine { cloud_type='flame', pow_min = 10, pow_max = 10, \
+                              delay = 10, size = 1, excl_rad = 0 }
+MAP
+xxxxxxxxxx
+x--xxxx
+x-!-xxxx
+xx-.xxxxx
+xxxx.^xxx
+xxxxx.xx
+xxx^.xx
+xxx@xx
+x
+ENDMAP
+
+NAME:   dreamdust_under_siege
+TAGS:   transparent no_monster_gen no_pool_fixup
+ORIENT: south
+DEPTH:  D:12-, Orc:1, Depths, !Depths:1
+WEIGHT: 5
+TILE:   c = wall_stone_smooth
+FTILE:  +'{([>-O = floor_cobble_blood
+# Monster Placement
+NSUBST: ' = 6=1 / 6=1' / '
+NSUBST: . = 10=2 / 10=2. / 10=3 / 10=3. / 2=4 / 3=4. / .
+# Theme variations
+: if you.in_branch("D") then
+TILE:   G : dngn_statue_angel
+KFEAT:  O = altar_zin / altar_elyvilon / altar_the_shining_one
+FTILE:  -O : floor_limestone / floor_w_marble
+MONS:   vault guard, human zombie / ogre zombie, \
+        human skeleton / ogre skeleton, necromancer
+: elseif you.in_branch("Orc") then
+SUBST:  ( = {, [ = >, G = I
+KFEAT:  > = escape_hatch_down
+KFEAT:  O = altar_beogh
+KPROP:  -O = bloody
+MONS:   orc / orc warrior, orc zombie / ogre zombie, \
+        orc skeleton / ogre skeleton, necromancer
+: else
+# Depths
+:    if crawl.coinflip() then
+# Holy defenders
+TILE:   G : dngn_statue_angel
+KFEAT:  O = altar_zin / altar_elyvilon / altar_the_shining_one
+FTILE:  -O : floor_limestone / floor_w_marble
+MONS:   angel / vault guard, zombie, simulacrum, lich / ancient lich w:3
+:    else
+# Unholy defenders
+TILE:   G : dngn_statue_imp / dngn_statue_demonic_bust
+KFEAT:  O = altar_yredelemnul / altar_kikubaaqudgha
+KPROP:  -O = bloody
+MONS:   ancient champion / vampire knight / lich, \
+        angel / nothing, vault guard, daeva / angel
+:    end
+: end
+MAP
+  ...........................
+ .............................
+...............................
```

Comment:
I'd recommend trimming the surrounding floor on `.` to only two squares deep around the entire perimeter of the vault. You only need a maximum of 45 squares in the case when your `NSUBST` for `.` places everything, which obviously wouldn't happen in practice, and and two squares of `.` surrounding the vault gives you significantly more than this. The smaller the vault is, the more likely it is to successfully place.